### PR TITLE
feat: Rotating Logs that can be Stored Locally by Default

### PR DESCRIPTION
**Pull request by davidterpay**
By default, sidecar logs will be written to `sidecar.log` with a max size of 100mb, 3 days of logs, 0 backups (i dont want to ballon this in case validators restart often), and info level logging (we can change once we reduce the amount of debug logs being written).

### CODE REVIEW

File: cmd/slinky/main.go [hunk 1]
```diff
@@ -90,11 +91,18 @@ func init() {
 		"info",
 		"Log level (debug, info, warn, error, dpanic, panic, fatal).",
 	)
+	rootCmd.Flags().StringVarP(
+		&fileLogLevel,
+		"file-log-level",
+		"",
+		"info",
```
Comment (davidterpay) [inline cmd/slinky/main.go:102]: can move this to debug once we purge and bubble up some of the logs.

Comment (aljo242) [reply cmd/slinky/main.go:102]: make linear task for this?

File: pkg/log/zap.go [hunk 2]
```diff
@@ -1,77 +1,101 @@
...
-	// outputPaths is the list of paths to write logs to. By default, logs are
-	// written to stderr.
-	outputPaths := []string{"stderr"}
-	if config.WriteTo != "" {
-		outputPaths = append(outputPaths, config.WriteTo)
+	// Setup the primary output to always include os.Stderr
+	stderrSyncer := zapcore.Lock(os.Stderr)
+
+	var fileCore zapcore.Core
+	if config.WriteTo != "" && config.WriteTo != "stderr" {
```
Comment (aljo242) [inline pkg/log/zap.go]: let's make "stderr" a `const`

**Review by aljo242 — APPROVED** [review]
lgtm w comment

**Pull request merged**